src/shims/unix/fs.rs

```diff
@@ -585,8 +585,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 );
             };
 
-            if mode != 0o666 {
-                throw_unsup_format!("non-default mode 0o{:o} is not supported", mode);
+            // currently we only support that the owner must always have
+            // read-write permissions and that none of the owner, group
+            // and other may have execute permissions.
+            if mode & 0o666 != mode | 0o600 {
```

Comment:
my brain broke trying to process this. Maybe we should start separating out the information and process it with exhaustive matches?

Reply:
To make this easier to digest, I can think of two flavors :)

```rust
            if !matches!(
                mode,
                0o600
                    | 0o602
                    | 0o604
                    | 0o606
                    | 0o620
                    | 0o622
                    | 0o624
                    | 0o626
                    | 0o640
                    | 0o642
                    | 0o644
                    | 0o646
                    | 0o660
                    | 0o662
                    | 0o664
                    | 0o666
            ) {
                throw_unsup_format!("mode 0o{:o} is not supported", mode);
            }
```

or

```rust
            let owner_read_write = 0o600;
            let group_read_write = 0o060;
            let other_read_write = 0o006;
            if mode & (owner_read_write | group_read_write | other_read_write)
                != mode | owner_read_write
            {
                throw_unsup_format!("mode 0o{:o} is not supported", mode);
            }
```

Which one do you prefer?

Reply:
p.s. another third option:

```rust
            let owner_read_write = 0o600;
            let execute = 0o111;
            if mode & owner_read_write != owner_read_write || mode & execute != 0
            {
                throw_unsup_format!("mode 0o{:o} is not supported", mode);
            }
```

Reply:
I was thinking about something along the lines of

```rust
            let owner = (mode >> 6) & 0o600;
            let group = (mode >> 3) & 0o060;
            let other = mode & 0o006;
            if do_some_logic_here {
                throw_unsup_format!("mode 0o{:o} is not supported", mode);
            }
```